public/app/plugins/datasource/loki/components/LokiContextUi.tsx

```diff
@@ -253,8 +252,8 @@ export function LokiContextUi(props: LokiContextUiProps) {
     };
   }, []);
 
-  // Currently we support adding of parser and showing parsed labels only if there is 1 parser
-  const showParsedLabels = origQuery && isQueryWithParser(origQuery.expr).parserCount === 1 && parsedLabels.length > 0;
+  // If there's any "parsedLabel", that also includes structured metadata, we show the parsed labels input
+  const showParsedLabels = parsedLabels.length > 0;
```

Comment:
```suggestion
  const showNonIndexedLabels = nonIndexedLabels.length > 0;
```

Could we change it to something like this - to cover also strucutred metadata?